backend/src/controllers/auth-controller.ts

```diff
@@ -62,4 +70,91 @@ export const loginUser = async (req: Request, res: Response) => {
     } catch (error: any) {
         res.status(500).json({ message: 'Server error', error: error.message });
     }
-}
\ No newline at end of file
+}
+
+export const postReset = async (req: Request, res: Response) => {
+    try {
+        crypto.randomBytes(32, (err, buf) => {
+           if (err){
+               console.log(err);
+           }
+           const token = buf.toString('hex');
+            const resetPassword = async ({req, res}: { req: any, res: any }) => {
```

Comment:
Make this one function -- `resetPassword` shouldn't be inside `postReset`.
Make sure to test that your code works as well -- while a call to `postReset` will say it succeeded, the logic within `resetPassword` is never called.